src/test/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudServiceTest.kt

```diff
@@ -265,12 +385,28 @@ class CosmasGoogleCloudServiceTest {
                 .newBuilder()
                 .setProjectId(projectId)
                 .build()
-
         return Pair(commitRecorder, commitRequest)
     }
 
     private fun commit(projectId: String = "0") {
         val (commitRecorder, commitRequest) = getStreamRecorderAndRequestForCommitVersions(projectId)
         this.service.commitVersion(commitRequest, commitRecorder)
     }
+
+    private fun addPatchToService(text: String, user: String, fileId: String, time: Long, projectId: String) {
+        val createPatchRecorder: StreamRecorder<CosmasProto.CreatePatchResponse> = StreamRecorder.create()
+        val newPatch = CosmasProto.Patch.newBuilder().setUserId(user).setText(text).setTimeStamp(time).build()
+        val newPatchRequest = CosmasProto.CreatePatchRequest
+                .newBuilder()
+                .setFileId(fileId)
+                .setProjectId(projectId)
+                .setPatch(newPatch)
+                .build()
+        this.service.createPatch(newPatchRequest, createPatchRecorder)
+    }
+
+    private fun checkCorrect(user: String, text: String, time: Long, ans: CosmasProto.Patch?) : Boolean {
```

Comment:
I recommend creating a new Patch from arguments (you can do it either directly in the test case code or create a helper method like this) and using `assertEquals(newPatch(userId, text, timestamp), list1[0])`

This is more self-describing way: the reader sees that test expects equality and doesn't need to learn what is "correctness" in test understanding.

Reply:
ok